test/e2e/basic/nodeport.go

```diff
@@ -66,37 +67,40 @@ func (n *NodePort) CreateResources() error {
-		Expect(createServiceWithNodeport(ctx, n.Client, n.namespace, n.serviceName, n.labels, 0)).To(Succeed(), fmt.Sprintf("Failed to create service %s", n.serviceName))
-		service, err := GetService(ctx, n.Client, n.namespace, n.serviceName)
-		Expect(err).To(Succeed())
-		Expect(len(service.Spec.Ports)).To(Equal(1))
-		n.nodePort = service.Spec.Ports[0].NodePort
-		_, err = GetAllService(ctx)
-		Expect(err).To(Succeed(), "fail to get service")
-	})
+	for _, ns := range *n.NSIncluded {
+		By(fmt.Sprintf("Creating service %s in namespaces %s ......\n", n.serviceName, ns), func() {
+			Expect(CreateNamespace(ctx, n.Client, ns)).To(Succeed(), fmt.Sprintf("Failed to create namespace %s", ns))
+			Expect(createServiceWithNodeport(ctx, n.Client, ns, n.serviceName, n.labels, 0)).To(Succeed(), fmt.Sprintf("Failed to create service %s", n.serviceName))
+			service, err := GetService(ctx, n.Client, ns, n.serviceName)
+			Expect(err).To(Succeed())
+			Expect(len(service.Spec.Ports)).To(Equal(1))
+			n.nodePort = service.Spec.Ports[0].NodePort
+			_, err = GetAllService(ctx)
+			Expect(err).To(Succeed(), "fail to get service")
+		})
+	}
 	return nil
 }
 
 func (n *NodePort) Destroy() error {
 	ctx, ctxCancel := context.WithTimeout(context.Background(), 60*time.Minute)
 	defer ctxCancel()
-	By(fmt.Sprintf("Start to destroy namespace %s......", n.NSBaseName), func() {
-		Expect(CleanupNamespacesWithPoll(ctx, n.Client, NodeportBaseName)).To(Succeed(),
-			fmt.Sprintf("Failed to delete namespace %s", n.NSBaseName))
-		Expect(WaitForServiceDelete(n.Client, n.namespace, n.serviceName, false)).To(Succeed(), "fail to delete service")
-		_, err := GetAllService(ctx)
-		Expect(err).To(Succeed(), "fail to get service")
-	})
-
-	n.namespaceToCollision = NodeportBaseName + "tmp"
-	By(fmt.Sprintf("Creating a new service which has the same nodeport as backed up service has in a new namespaces for nodeport collision ...%s\n", n.namespaceToCollision), func() {
-		Expect(CreateNamespace(ctx, n.Client, n.namespaceToCollision)).To(Succeed(), fmt.Sprintf("Failed to create namespace %s", n.namespaceToCollision))
-		Expect(createServiceWithNodeport(ctx, n.Client, n.namespaceToCollision, n.serviceName, n.labels, n.nodePort)).To(Succeed(), fmt.Sprintf("Failed to create service %s", n.serviceName))
-		_, err := GetAllService(ctx)
-		Expect(err).To(Succeed(), "fail to get service")
-	})
+	for _, ns := range *n.NSIncluded {
+		By(fmt.Sprintf("Start to destroy namespace %s......", n.CaseBaseName), func() {
+			Expect(CleanupNamespacesWithPoll(ctx, n.Client, NodeportBaseName)).To(Succeed(),
+				fmt.Sprintf("Failed to delete namespace %s", n.CaseBaseName))
+			Expect(WaitForServiceDelete(n.Client, ns, n.serviceName, false)).To(Succeed(), "fail to delete service")
+			_, err := GetAllService(ctx)
+			Expect(err).To(Succeed(), "fail to get service")
+		})
+		namespaceToCollision := ns + "-tmp"
```

Comment:
I like to have variable namespaceToCollision as a case scope one.

Reply:
I've added it back into NodePort